android/java/org/chromium/chrome/browser/crypto_wallet/activities/AssetDetailActivity.java

```diff
@@ -111,6 +112,10 @@ protected void triggerLayoutInflation() {
         }
         mExecutor = Executors.newSingleThreadExecutor();
         mHandler = new Handler(Looper.getMainLooper());
+        BraveActivity activity = BraveActivity.getBraveActivity();
+        if (activity != null) {
```

Comment:
Considering the `mWalletModel` field is used in the `observeOnce` method below, it's best to assert (i.e. `assert activity != null`) to be sure the Brave activity will never be `null`. Very unlikely to happen if I've understood correctly, but asserting we make sure we don't access a null object.

Reply:
no, without assert the app will crash anyway so it is not required here.

Reply:
So if we are not preventing the app from crashing, what's the point of this check? 🤔

Reply:
how assert will help here to prevent a crash?

Reply:
maybe we can just add a check on `mWalletModel == null` below and do nothing for now. That should go away when we move everything out of the activity as we discussed in a DM

Reply:
I can add the null check though as per the logs, that seems to be the issue with the internal `BraveWalletService_Internal.MANAGER` object being null and outer objects are initialised properly. Gonna add the check.